File: 1552-build-an-array-with-stack-operations/build-an-array-with-stack-operations.py
class Solution(object):
    def buildArray(self, target, n):
        """
        :type target: List[int]
        :type n: int
        :rtype: List[str]
        """
        ops = []
        idx = 0
        for i in range(1,n+1):
            ops.append("Push")
            if(idx < len(target) and target[idx]==i):
                idx+=1
            else:
                ops.append("Pop")
            
            if(idx == len(target)):
                break
        
        return ops

File: 1552-build-an-array-with-stack-operations/build-an-array-with-stack-operations.cpp
class Solution {
public:
    vector<string> buildArray(vector<int>& target, int n) {
        vector<string> ops;
        int idx = 0;
        for(int i=1;i<=n;i++){
            ops.push_back("Push");
            if(idx < target.size() && target[idx]==i){
                idx++;
            }else
            {
                ops.push_back("Pop");
            }
            if(idx == target.size())
            break;
        }
        return ops;
    }
};